client/state/account-recovery/reset/reducer.js

```diff
@@ -83,11 +86,30 @@ const validate = combineReducers( {
 	} ),
 } );
 
+const resetPassword = combineReducers( {
+	isRequesting: createReducer( false, {
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST ]: stubTrue,
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST_SUCCESS ]: stubFalse,
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST_ERROR ]: stubFalse,
+	} ),
+	succeeded: createReducer( false, {
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST ]: stubFalse,
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST_SUCCESS ]: stubTrue,
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST_ERROR ]: stubFalse,
+	} ),
+	error: createReducer( null, {
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST ]: () => null,
+		[ ACCOUNT_RECOVERY_RESET_PASSWORD_REQUEST_SUCCESS ]: () => null,
```

Comment:
Maybe we should reset these to the default `false` instead of `null`. Your tests below are checking for a `false` value